packages/grid/x-data-grid/src/components/panel/GridColumnsPanel.tsx

```diff
@@ -136,9 +142,15 @@ function GridColumnsPanel(props: GridColumnsPanelProps) {
     );
   }, [sortedColumns, searchValue]);
 
+  const firstSwitchRef = React.useRef<HTMLInputElement>(null);
+
   React.useEffect(() => {
-    searchInputRef.current!.focus();
-  }, []);
+    if (autoFocusSearchField) {
+      searchInputRef.current!.focus();
+    } else if (firstSwitchRef.current && typeof firstSwitchRef.current.focus === 'function') {
+      firstSwitchRef.current.focus();
+    }
+  }, [autoFocusSearchField]);
```

Comment:
It would be great if we focus the switch only if the column is hideable :)
To illustrate the problem see this demo: https://codesandbox.io/s/winter-snow-2q8rb5?file=/demo.tsx
The first column is not hideable, so once you open columns panel, neither input search nor switch input are focused.

```suggestion
  }, [autoFocusSearchField]);
  
  let firstHideableColumnFound = false;
  const isFirstHideableColumn = (column: GridStateColDef) => {
    if (firstHideableColumnFound === false && column.hideable !== false) {
      firstHideableColumnFound = true;
      return true;
    }
    return false;
  };
```